MapboxCoreNavigation/RouteController.swift

```diff
@@ -541,6 +543,28 @@ extension RouteController: CLLocationManagerDelegate {
         let radius = max(reroutingTolerance, location.horizontalAccuracy + RouteControllerUserLocationSnappingDistance)
         let isCloseToCurrentStep = newLocation.isWithin(radius, of: routeProgress.currentLegProgress.currentStep)
         
+        
+        // Check to see if the user is moving away from the maneuver.
+        // Here, we store an array of distances. If the current distance is greater than the last distance,
+        // add it to the array. If the array grows larger than x, reroute the user.
+        if let coordinates = routeProgress.currentLegProgress.currentStep.coordinates {
+            let userDistanceToManeuver = distance(along: coordinates, from: location.coordinate)
+            
+            guard previousDistancesFromManeuver.count <= 3 else {
+                resetPreviousDistanceArray()
+                return false
+            }
+            
+            if previousDistancesFromManeuver.isEmpty {
+                previousDistancesFromManeuver.append(userDistanceToManeuver)
+            } else if let lastSpeed = previousDistancesFromManeuver.last, userDistanceToManeuver > lastSpeed {
+                previousDistancesFromManeuver.append(userDistanceToManeuver)
```

Comment:
`RouteStepProgress.userDistanceToManeuverLocation` already tracks the most recent distance, so all this class needs is to keep a counter of increasing distances: `countOfRecentBackwardsUpdates` or something along those lines.